app/qml/main.qml

```diff
@@ -30,8 +30,14 @@ ApplicationWindow{
     height: 768
 
     // Save window properties automatically
-    onXChanged: appSettings.x = x
-    onYChanged: appSettings.y = y
+    onXChanged: {
+        appSettings.x = x
+        appSettings.pixelRatio = Window.devicePixelRatio
```

Comment:
In my tests with Qt 5.12 this always returns undefined. Did you mean something like "terminalWindow.screen.devicePixelRatio"?